packages/wdio-browserstack-service/src/insights-handler.ts

```diff
@@ -430,6 +452,37 @@ class _InsightsHandler {
         }
     }
 
+    getTestRunId(context: any): string|undefined {
+        if (!context) {return}
+
+        if (context.currentTest) {
+            const uniqueIdentifier = getUniqueIdentifier(context.currentTest, this._framework)
+            return this._tests[uniqueIdentifier] && this._tests[uniqueIdentifier].uuid
+        }
+
+        if (!context.test) {
+            return
+        }
+
+        const getTestRunIdFromSuite = (parent: any): string|undefined => {
```

Comment:
Let's add a null check on parent arg & return null so that it doesn't break + recursive tree ends.